### PR TITLE
[Sprint: 45] XD-2833: Document MongoDB source, add split

### PR DESCRIPTION
**Pull request by ericbottard**

### CODE REVIEW

File: spring-xd-module-spi/src/main/java/org/springframework/xd/module/options/mixins/FromMongoDbOptionMixin.java [hunk 1]
```diff
@@ -107,4 +115,19 @@ public String getQuery() {
 		return this.query;
 	}
 
+	public boolean isSplit() {
+		return split;
+	}
+
+	@ModuleOption("whether to split the SQL result as individual messages")
```
Comment (ericbottard) [inline spring-xd-module-spi/src/main/java/org/springframework/xd/module/options/mixins/FromMongoDbOptionMixin.java]: Will need to change this awful copy/paste error

File: spring-xd-module-spi/src/main/java/org/springframework/xd/module/options/mixins/FromMongoDbOptionMixin.java [hunk 2]
```diff
@@ -40,7 +45,10 @@
 	private int pollRate = 1000;
 	
 	private int maxMessages = 1;
-	
+
+	private boolean split = true;
```
Comment (garyrussell) [inline spring-xd-module-spi/src/main/java/org/springframework/xd/module/options/mixins/FromMongoDbOptionMixin.java]: We should add a comment to XD-2794 to give the contributor a heads-up that the source now splits by default.

Comment (ericbottard) [inline spring-xd-module-spi/src/main/java/org/springframework/xd/module/options/mixins/FromMongoDbOptionMixin.java]: Indeed. Will do.

File: modules/source/mongodb/config/mongodb.xml [hunk 3]
```diff
@@ -17,9 +19,15 @@
 		username="${username}" password="${password}" />
 
 	<int-mongodb:inbound-channel-adapter
-		id="inbound" channel="output" query="${query}" collection-name="${collectionName}">
+		id="inbound" channel="aggregates" query="${query}" collection-name="${collectionName}">
 		<int:poller fixed-rate="${pollRate}" max-messages-per-poll="${maxMessages}" />
```
Comment (garyrussell) [inline modules/source/mongodb/config/mongodb.xml]: While not really a "documentation" issue, we should use `fixedDelay` not `pollRate` in this source to be consistent with the `time` and `trigger` sources.

For the same reason, we should also consider adding `initialDelay` and `timeUnit` - see https://github.com/spring-projects/spring-xd/pull/1384/files and the `PeriodicTriggerMixin`.

**Comment by ericbottard** [comment]
I created https://jira.spring.io/browse/XD-2846 to track the `initialDelay` thing, as there are several modules that could benefit from this.
I suggest merging this PR if this is the only outstanding issue

**Comment by ericbottard** [comment]
Actually, I'm applying it to this PR to avoid `pollRate` going in the wild

File: modules/source/mongodb/config/mongodb.xml [hunk 4]
```diff
@@ -17,9 +19,21 @@
...
+	</bean>
+
+	<beans profile="use-splitter">
+		<int:splitter input-channel="aggregates" output-channel="output"/>
+	</beans>
+	<beans profile="dont-use-splitter">
+		<int:bridge input-channel="aggregates" output-channel="output"/>
+	</beans>
+
 </beans>
```
Comment (garyrussell) [inline modules/source/mongodb/config/mongodb.xml]: Needs a newline; will fix on merge

**Comment by garyrussell** [comment]
LGTM; minor polishing [here](https://github.com/garyrussell/spring-xd/commit/db8ac282198991f7ec9dc6627190ab81e1cc2bd2) for your approval before I merge.

**Comment by garyrussell** [comment]
Merged as 3c31165cbdd2760373b40ecaffe4626682c82a13

**Pull request closed without merging**